acvm-repo/acir/src/circuit/opcodes/black_box_function_call.rs

```diff
@@ -83,33 +83,54 @@ impl<F: std::fmt::Display> std::fmt::Display for FunctionInput<F> {
 
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Hash)]
 pub enum BlackBoxFuncCall<F> {
+    /// Ciphers (encrypts) the provided plaintext using AES128 in CBC mode,
+    /// padding the input using PKCS#7.
+    /// - inputs: byte array `[u8; N]`
+    /// - iv: initialization vector `[u8; 16]`
+    /// - key: user key `[u8; 16]`
+    /// - outputs: byte vector `[u8]` of length `input.len() + (16 - input.len() % 16)`
     AES128Encrypt {
         inputs: Vec<FunctionInput<F>>,
         iv: Box<[FunctionInput<F>; 16]>,
         key: Box<[FunctionInput<F>; 16]>,
         outputs: Vec<Witness>,
     },
-    AND {
-        lhs: FunctionInput<F>,
-        rhs: FunctionInput<F>,
-        output: Witness,
-    },
-    XOR {
-        lhs: FunctionInput<F>,
-        rhs: FunctionInput<F>,
-        output: Witness,
-    },
-    RANGE {
-        input: FunctionInput<F>,
-    },
-    Blake2s {
-        inputs: Vec<FunctionInput<F>>,
-        outputs: Box<[Witness; 32]>,
-    },
-    Blake3 {
-        inputs: Vec<FunctionInput<F>>,
-        outputs: Box<[Witness; 32]>,
-    },
+    /// Performs the bitwise AND of `lhs` and `rhs`. `bit_size` must be the same for
+    /// both inputs.
+    /// - lhs: (witness, bit_size)
+    /// - rhs: (witness, bit_size)
+    /// - output: a witness whose value is constrained to be lhs AND rhs, as
+    ///   bit_size bit integers
+    AND { lhs: FunctionInput<F>, rhs: FunctionInput<F>, output: Witness },
+    /// Performs the bitwise XOR of `lhs` and `rhs`. `bit_size` must be the same for
+    /// both inputs.
+    /// - lhs: (witness, bit_size)
+    /// - rhs: (witness, bit_size)
+    /// - output: a witness whose value is constrained to be lhs XOR rhs, as
+    ///   bit_size bit integers
+    XOR { lhs: FunctionInput<F>, rhs: FunctionInput<F>, output: Witness },
+    /// Range constraint to ensure that a witness
+    /// can be represented in the specified number of bits.
+    /// - input: (witness, bit_size)
+    RANGE { input: FunctionInput<F> },
+    /// Computes the Blake2s hash of the inputs, as specified in
+    /// https://tools.ietf.org/html/rfc7693
+    /// - inputs are a byte array, i.e a vector of (witness, 8)
+    /// - output is a byte array of length 32, i.e. an array of 32
+    ///   (witness, 8), constrained to be the blake2s of the inputs.
+    Blake2s { inputs: Vec<FunctionInput<F>>, outputs: Box<[Witness; 32]> },
+    /// Computes the Blake3 hash of the inputs
+    /// - inputs are a byte array, i.e a vector of (witness, 8)
+    /// - output is a byte array of length 32, i.e an array of 32
+    ///   (witness, 8), constrained to be the blake3 of the inputs.
+    Blake3 { inputs: Vec<FunctionInput<F>>, outputs: Box<[Witness; 32]> },
+    /// Verifies a ECDSA signature over the secp256k1 curve.
```

Comment:
We should document that we require that s has been normalized for verification to succeed.

Reply:
Actually, can you add a doc comment to the ecdsa functions in the stdlib with the same thing?

Reply:
Had to use regular comments in the stdlib https://github.com/noir-lang/noir/issues/7711